envelope.proto

```diff
@@ -32,4 +34,19 @@ message Signature {
   // *Unauthenticated* hint identifying which public key was used.
   // OPTIONAL.
   string keyid = 2;
+
+  // *Unauthenticated* field that can record additional information pertaining
+  // to the signature.
+  // OPTIONAL.
+  Extension extension = 3;
+}
+
+message Extension {
+  // Unambiguously identifies how to interpret extension fields.
+  // REQUIRED.
+  string type_ = 1;
```

Comment:
Does json serialization strip the _?  If not I think you can use the `json_name` extension? https://protobuf.dev/programming-guides/proto3/#json

Reply:
I'm not used to seeing fields ending in `_`, can we remove?

Reply:
Agreed. If the issue here is that `type` might be a keyword in some languages, perhaps we could make it `kind`?